src/t8_geometry/t8_geometry_implementations/t8_geometry_occ.cxx

```diff
@@ -0,0 +1,726 @@
+        interpolated_coords[dim] += face_displacement_from_edges[dim];
+      }
+
+      for (int dim = 0; dim < 2; ++dim) {
+        /* Correct the interpolated parameters with the parameter displacement generated by the edges */
+        interpolated_surface_params[dim] +=
+          surface_parameter_displacement_from_edges[dim];
+      }
+
+      /* *INDENT-OFF* */
+      /* Retrieve the surface of the edge */
+      T8_ASSERT (faces[i_faces] <= occ_shape_face_map.Size ());
+      surface =
+        BRep_Tool::Surface (TopoDS::Face (occ_shape_face_map.FindKey (faces[i_faces])));
+      /* *INDENT-ON* */
+
+      /* Check if surface is valid */
+      T8_ASSERT (!surface.IsNull ());
+
+      /* Compute point on surface with interpolated parameters */
+      surface->D0 (interpolated_surface_params[0],
+                   interpolated_surface_params[1], pnt);
+
+      /* Compute the displacement between surface and interpolated coords, scale them with the appropriate ref_coord 
+       * and add them to the out_coords. */
+      if (i_faces % 2 == 0) {
+        out_coords[0]
+          += (pnt.X () - interpolated_coords[0])
+          * (1 - ref_coords[face_normal_direction]);
+        out_coords[1]
+          += (pnt.Y () - interpolated_coords[1])
+          * (1 - ref_coords[face_normal_direction]);
+        out_coords[2]
+          += (pnt.Z () - interpolated_coords[2])
+          * (1 - ref_coords[face_normal_direction]);
+      }
+      else {
+        out_coords[0]
+          += (pnt.X () - interpolated_coords[0])
+          * ref_coords[face_normal_direction];
+        out_coords[1]
+          += (pnt.Y () - interpolated_coords[1])
+          * ref_coords[face_normal_direction];
+        out_coords[2]
+          += (pnt.Z () - interpolated_coords[2])
+          * ref_coords[face_normal_direction];
+      }
+    }
+  }
+}
+
+void
+t8_geometry_occ::t8_geom_evalute_jacobian (t8_cmesh_t cmesh,
+                                           t8_gloidx_t gtreeid,
+                                           const double
+                                           *ref_coords,
+                                           double *jacobian_out) const
+{
+  double              h = 1e-9;
+  double              in1[3], in2[3];
+  double              out1[3], out2[3];
+  for (int dim = 0; dim < 3; ++dim) {
+    memcpy (in1, ref_coords, sizeof (double) * 3);
+    memcpy (in2, ref_coords, sizeof (double) * 3);
+
+    if (ref_coords[dim] < h) {
+      in2[dim] += ref_coords[dim] + h;
+    }
+    else if (ref_coords[dim] > 1 - h) {
+      in1[dim] -= h;
+    }
+    else {
+      in1[dim] -= 0.5 * h;
+      in2[dim] += 0.5 * h;
+    }
+    t8_geometry_occ::t8_geom_evaluate (cmesh, gtreeid, in1, out1);
+    t8_geometry_occ::t8_geom_evaluate (cmesh, gtreeid, in2, out2);
+    for (int dim2 = 0; dim2 < 3; ++dim2) {
+      jacobian_out[dim * 3 + dim2] = out1[dim2] - out2[dim2];
```

Comment:
Isnt a division by h missing?

`f' = (f(x+h) - f(x))/h`

i only see
`f' = f(x+h) - f(h)`

Reply:
Yes, I just forgot that.
resolved in 2c4a4e5f